### PR TITLE
Feature/cart

### PR DESCRIPTION
**Pull request by properweb**

### CODE REVIEW

File: Modules/Cart/Http/Controllers/CartController.php [hunk 1]
```diff
@@ -21,34 +22,94 @@ public function __construct(CartService $cartService)
...
     {
+        $user = auth()->user();
+        $cart = Cart::where('id', $request->id)->first();
+        if ($user->cannot('delete', $cart)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'You are not authorized !',
+                'data' => ""
+            ]);
+        }
```
Comment (SonaKhachatryan1995) [inline Modules/Cart/Http/Controllers/CartController.php:83]: for $response = $this->cartService->delete($request->id,$request->user_id); do not $request->user_id. From front end never send user_id.The code should be like this

$response = $this->cartService->delete($request->id)

File: Modules/Cart/Http/Services/CartService.php [hunk 2]
```diff
@@ -261,108 +277,123 @@ public function fetch($requestData): array
...
      */
-    public function delete($id, $user_id): array
+    public function update($request): array
     {
-        $cart = Cart::where('id', $id)->where('user_id', $user_id)->first();
-        if (empty($cart)) {
+        $carts = $request->cart;
+        $error = 0;
+        foreach ($carts as $v) {
+            $cart = Cart::where('id', $v['id'])->where('user_id', $v['user_id'])->first();
```
Comment (SonaKhachatryan1995) [inline Modules/Cart/Http/Services/CartService.php]: Should be $cart = Cart::where('id', $v['id'])->first();
do not use user_id from front end, use auth()->user()->id.

File: Modules/Cart/Http/Services/CartService.php [hunk 3]
```diff
@@ -261,108 +277,123 @@ public function fetch($requestData): array
...
-            'country' => $country,
-            'street' => $street,
-            'suite' => $suite,
-            'state' => $state,
-            'town' => $town,
-            'zip' => $zip,
-            'phoneCode' => $phoneCode
-        );
-        Cart::where('id', $id)->update($data);
+        $cart = Cart::where('id', $id)->where('user_id', $user_id)->first();
```
Comment (SonaKhachatryan1995) [inline Modules/Cart/Http/Services/CartService.php]: Should be $cart = Cart::where('id', $id)->first() or Cart::find($id).
do not use user_id from front end, use auth()->user()->id.

File: Modules/Cart/Http/Services/CartService.php [hunk 4]
```diff
@@ -261,108 +277,123 @@ public function fetch($requestData): array
...
-     * @param array $cartData
+     * User can delete product from his cart
+     *
+     * @param $id
+     * @param $user_id
      * @return array
      */
-
-    public function update(array $cartData): array
+    public function delete($id, $user_id): array
```
Comment (SonaKhachatryan1995) [inline Modules/Cart/Http/Services/CartService.php]: Remove $user_id parameter

File: Modules/Cart/Http/Services/CartService.php [hunk 5]
```diff
@@ -261,108 +277,123 @@ public function fetch($requestData): array
 
 
         $cartCount = 0;
-        $cart_arr = [];
+        $cartArr = [];
```
Comment (SonaKhachatryan1995) [inline Modules/Cart/Http/Services/CartService.php:277]: $id = $requestData->id;
        $user = User::find($id); 
this 2 lines should be replaced with fallowing: 
        $user = auth()->user();
        $id = $user->id;

File: Modules/Cart/Http/Services/CartService.php [hunk 6]
```diff
@@ -232,26 +246,28 @@ public function add(array $request): array
...
+
+                    return ['res' => false, 'msg' => 'Stock not sufficient!.', 'data' => ""];
+
                 }
             }
             $cart->amount = $cart->price * $cart->quantity;
             $cart->type = $productType;
             $cart->save();
-            $wishlist = Wishlist::where('user_id', $request['user_id'])->where('cart_id', null)->update(['cart_id' => $cart->id]);
+            Wishlist::where('user_id', $request->user_id)->where('cart_id', null)->update(['cart_id' => $cart->id]);
```
Comment (SonaKhachatryan1995) [inline Modules/Cart/Http/Services/CartService.php]: Wishlist::where('user_id', $request->user_id)->where('cart_id', null)->update(['cart_id' => $cart->id]); should be changed to
  Wishlist::where('user_id', auth()->user()->id)->where('cart_id', null)->update(['cart_id' => $cart->id]);

File: Modules/Cart/Http/Services/CartService.php [hunk 7]
```diff
@@ -119,56 +123,62 @@ public function add(array $request): array
...
+                    return ['res' => false, 'msg' => 'Stock not sufficient!.', 'data' => ""];
+
                 }
             }
             $alreadyCart->save();
         } else {
             $cart = new Cart;
 
-            $cart->user_id = $request['user_id'];
+            $cart->user_id = $request->user_id;
```
Comment (SonaKhachatryan1995) [inline Modules/Cart/Http/Services/CartService.php]: $cart->user_id = $request->user_id; should be changed to
$cart->user_id = auth()->user()->id;

File: Modules/Cart/Http/Services/CartService.php [hunk 8]
```diff
@@ -19,71 +19,76 @@ class CartService
...
                 $productType = 'PREPACK';
-                $variantId = $request['prepack_id'];
+                $variantId = $preID;
+
             }
         }
-
-        if (!empty($request['variant_id']) && $variant) {
-            $alreadyCart = Cart::where('user_id', $request['user_id'])->where('order_id', null)->where('product_id', $product->id)->where('variant_id', $variantId)->where('type', $productType)->first();
+        if (!empty($request->prepack_id)) {
```
Comment (SonaKhachatryan1995) [inline Modules/Cart/Http/Services/CartService.php:66]: if (!empty($request->prepack_id)) {
            $alreadyCart = Cart::where('user_id', $request->user_id)->where('order_id', null)->where('product_id', $product->id)->where('reference', $getVariant)->where('type', $productType)->first();
        } else {
            if (!empty($getVariant) && !empty($variant)) {
                $alreadyCart = Cart::where('user_id', $request->user_id)->where('order_id', null)->where('product_id', $product->id)->where('variant_id', $variantId)->where('type', $productType)->first();
            } else {
                $alreadyCart = Cart::where('user_id', $request->user_id)->where('order_id', null)->where('product_id', $product->id)->where('type', $productType)->first();
            }
        }

replace all $request->user_id to auth()->user()->id

**Review by SonaKhachatryan1995 — CHANGES_REQUESTED** [review]

File: Modules/Cart/Http/Controllers/CartController.php (hunk 1 above)
Comment (arijit1111) [reply Modules/Cart/Http/Controllers/CartController.php:83]: Done

File: Modules/Cart/Http/Services/CartService.php (hunk 2 above)
Comment (arijit1111) [reply Modules/Cart/Http/Services/CartService.php]: Done

File: Modules/Cart/Http/Services/CartService.php (hunk 3 above)
Comment (arijit1111) [reply Modules/Cart/Http/Services/CartService.php]: Done

File: Modules/Cart/Http/Services/CartService.php (hunk 4 above)
Comment (arijit1111) [reply Modules/Cart/Http/Services/CartService.php]: Done

File: Modules/Cart/Http/Services/CartService.php (hunk 5 above)
Comment (arijit1111) [reply Modules/Cart/Http/Services/CartService.php:277]: Done

File: Modules/Cart/Http/Services/CartService.php (hunk 6 above)
Comment (arijit1111) [reply Modules/Cart/Http/Services/CartService.php]: Done

File: Modules/Cart/Http/Services/CartService.php (hunk 7 above)
Comment (arijit1111) [reply Modules/Cart/Http/Services/CartService.php]: Done

File: Modules/Cart/Http/Services/CartService.php (hunk 8 above)
Comment (arijit1111) [reply Modules/Cart/Http/Services/CartService.php:66]: Done

File: Modules/Cart/Http/Services/CartService.php (hunk 5 above)
Comment (SonaKhachatryan1995) [reply Modules/Cart/Http/Services/CartService.php:277]: Remove $requestData parameter, it's not needed.............also there is unnecessary spaces in the begining of function remove them as well.

File: Modules/Cart/Http/Services/CartService.php (hunk 6 above)
Comment (SonaKhachatryan1995) [reply Modules/Cart/Http/Services/CartService.php]: Variable name should be $userId not $user_id according to PSR standard

File: Modules/Cart/Http/Services/CartService.php (hunk 7 above)
Comment (SonaKhachatryan1995) [reply Modules/Cart/Http/Services/CartService.php]: Variable name should be $userId not $user_id according to PSR standard

File: Modules/Cart/Http/Services/CartService.php (hunk 5 above)
Comment (arijit1111) [reply Modules/Cart/Http/Services/CartService.php:277]: Done

File: Modules/Cart/Http/Services/CartService.php (hunk 6 above)
Comment (arijit1111) [reply Modules/Cart/Http/Services/CartService.php]: Done

File: Modules/Cart/Http/Services/CartService.php (hunk 7 above)
Comment (arijit1111) [reply Modules/Cart/Http/Services/CartService.php]: Done

File: Modules/Cart/Http/Services/CartService.php (hunk 8 above)
Comment (arijit1111) [reply Modules/Cart/Http/Services/CartService.php:66]: Done

**Review by SonaKhachatryan1995 — APPROVED** [review]

**Pull request merged**